config/v1/types_insights.go

```diff
@@ -0,0 +1,72 @@
+package v1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+//
+// Insights holds cluster-wide information about Insights.
+//
+// Compatibility level 1: Stable within a major release for a minimum of 12 months or 3 minor releases (whichever is longer).
+// +openshift:compatibility-gen:level=1
+type Insights struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// spec holds user settable values for configuration
+	// +kubebuilder:validation:Required
+	// +required
+	Spec InsightsSpec `json:"spec"`
+	// status holds observed values from the cluster. They may not be overridden.
+	// +optional
+	Status InsightsStatus `json:"status"`
+}
+
+type InsightsSpec struct {
+	// GatherConfig spec attribute includes all the configuration options related to
+	// gathering of the Insights archive and its uploading to the ingress.
+	// +optional
+	GatherConfig *GatherConfig `json:"gatherConfig,omitempty"`
+}
+
+type InsightsStatus struct {
+}
+
+type GatherConfig struct {
+	// DataPolicy allows user to enable additional global obfuscation of the IP addresses and base domain
+	// in the Insights archive data.
+	// +kubebuilder:default=NoPolicy
+	DataPolicy DataPolicy `json:"dataPolicy"`
+	// ForceGatherReason enables user to force Insights data gathering by setting a new reason.
+	// When there is some gathering in the progress then it is interrupted.
+	// When all the gatherers are deactivated by the `DisabledGatherers`, nothing happens.
+	// When the forced gathering is finished then the value is cleared.
+	ForceGatherReason string `json:"forceGatherReason"`
+	// List of gatherers to be excluded from the gathering. All the gatherers can be disabled by providing "all" value.
+	// If all the gatherers are disabled, the Insights operator does not gather any data.
+	// The particular gatherers IDs can be found at https://github.com/openshift/insights-operator/blob/master/docs/gathered-data.md.
```

Comment:
A note from the enhancement, this link is likely to fail over time, can we list the gatherers that could be configured within the field documentation?

Reply:
Although desirable, the problem might be that we don't keep the gatherers as part of the API (as I understand it) so the two can quickly get out of sync. @tremes is there a better way how a user can get a list of gatherers from a running cluster, for example?

Reply:
Hmm there's no way how to get a list of gatherers from running cluster right now. Yes the list of gatherers (gathered data) changes over time, but I don't have any other suggestion than the mentioned link..... I will try to think about it more.